zschema/compounds.py

```diff
@@ -192,8 +192,11 @@ def validate(self, value):
             raise DataValidationException("record is not a dict", str(value))
         for subkey, subvalue in value.items():
             if subkey not in self.definition:
-                raise DataValidationException("%s is not a valid subkey of root",
-                                              subkey)
+                port = Port(subkey)
```

Comment:
It sounds like this should be fixed by defining `__eq__` on Keyable differently. And it actually looks like this was already attempted at one point:

https://github.com/zmap/zschema/blob/master/zschema/keys.py#L13

We should understand why this isn't working.